lib/irida/pipeline.rb

```diff
@@ -21,5 +23,36 @@ def initialize(entry, version, schema_loc, schema_input_loc)
       @schema_loc = schema_loc
       @schema_input_loc = schema_input_loc
     end
+
+    def workflow_params # rubocop:disable Metrics/AbcSize,Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+      nextflow_schema = JSON.parse(schema_loc.read)
+      workflow_params = {}
+
+      nextflow_schema['definitions'].each do |key, definition|
+        next unless show_section?(definition['properties'])
+
+        workflow_params[key] = { title: definition['title'], description: definition['description'], properties: {} }
+
+        definition['properties'].each do |name, property|
```

Comment:
If you move line 36 through 46 into its own method under private you can get rid of the cyclomatic and perceived complexity rubocop disables. I think it would make sense to move it into its own method as its setting the workflow_params based on each property in definition['properties']